scripts/browser-specs.js

```diff
@@ -48,59 +65,110 @@ request({
         })
         .forEach(function(spec) {
-            ref = convertSpec(spec);
-            
-            var id = spec.shortname;
-            var uppercaseId = id.toUpperCase();
-
-            // Spec may already exist in Specref under another name
-            var rv = bibref.reverseLookup([ref.href])[ref.href];
-            if (rv) {
+            // Add an entry to browser-specs.json for the spec, unless Specref
+            // already has an entry somewhere else for a spec with the same URL.
+            const uppercaseId = spec.shortname.toUpperCase();
+            const rv = bibref.reverseLookup([spec.url])[spec.url];
+            if (rv && !current[rv.id]) {
                 if (current[uppercaseId] &&
-                        rv.id.toUpperCase() != uppercaseId &&
-                        // avoid inadvertently catching drafts.
-                        bibref.normalizeUrl(rv.href) == bibref.normalizeUrl(ref.href)) {
-                    // Make the existing entry in browser-specs.json an alias
-                    // of the other one
+                        rv.id.toUpperCase() !== uppercaseId) {
+                    // There was an entry for this spec in browser-specs.json
+                    // but Specref has an entry for a spec with the same URL
+                    // somewhere else. Let's give that other entry priority and
+                    // mark the browser-specs entry as an alias of it.
                     updatedList[uppercaseId] = { aliasOf: rv.id };
                     return;
                 }
                 else if (!current[uppercaseId]) {
-                    // Entry did not exist in browser-specs.json, let's not
-                    // create one
+                    // There is an entry for this spec in Specref somewhere else
+                    // and we had not yet created a browser-specs.json entry.
+                    // Let's not.
                     return;
                 }
             }
 
-            if (!(uppercaseId in refs)) {
-                updatedList[uppercaseId] = ref;
-            } else {
-                var existingRef = refs[uppercaseId];
-                while (existingRef.aliasOf) { existingRef = refs[existingRef.aliasOf]; }
-                if (!("source" in existingRef) || existingRef.source == SOURCE ||
-                        bibref.normalizeUrl(ref.href) == bibref.normalizeUrl(existingRef.href)) {
-                    updatedList[uppercaseId] = ref;
-                } else {
-                    throw new Error("cannot override " + uppercaseId);
+            // No entry in Specref, let's add one
+            updatedList[uppercaseId] = convertSpec(spec);
+        });
+
+    // For continuity, add back aliases that we may have created earlier on,
+    // unless the targeted entry no longer exists (that should never happen
+    // though). Also add back entries that used to exist but that are now known
+    // under another name.
```

Comment:
FWIW, it's perfectly fine to throw an error in such cases, ideally that's clear enough to explain why this is weird and should never happen. Throwing in the update script only means that auto-update is broken. Nothing else.